### PR TITLE
Avoid float formatting, which reduces binary size by 30k (5%)

### PR DESCRIPTION
**Pull request by malbarbo**

### CODE REVIEW

File: src/editor.rs [hunk 1]
```diff
@@ -124,12 +124,16 @@ impl StatusMessage {
...
-    // unwrap(): quo_rem is never empty (since `successors` has an initial value), so _.last()
-    // cannot be None
-    let ((q, r), prefix) = quo_rem.zip(&["", "k", "M", "G", "T", "P", "E", "Z"]).last().unwrap();
-    format!("{:.2$}{}B", q as f32 + r as f32 / 1024., prefix, p = if *prefix == "" { 0 } else { 2 })
+    if n < 1024 {
+        return format!("{}B", n);
+    }
+    // i is the largest value such that 1024 ^ i < n
+    // To find i we compute the smallest b such that n <= 1024 ^ b and subtract -1 from b
+    let i = ((64 - n.leading_zeros()) as f64 / 10.0).ceil() as usize - 1;
```
Comment (ilai-deutel) [inline src/editor.rs:132]: I think this is equivalent and doesn't require casting to `f64`:

```suggestion
    let i = ((64 - n.leading_zeros() + 9) / 10) as usize - 1;
```

File: src/editor.rs [hunk 2]
```diff
@@ -124,12 +124,16 @@ impl StatusMessage {
...
+    // Compute the size multiplied by 100 as u64. We get two decimal places
+    // without using float formatting -- which reduces de binary size
+    let q = (100.0 * n as f64 / (1024 << ((i - 1) * 10)) as f64).round() as u64;
```
[the remark this answers is not in the document]
Comment (malbarbo) [reply src/editor.rs]: The tests looked odd to me, now I know why! Rounding down is the correct thing to do.

File: src/editor.rs (hunk 1 above)
Comment (malbarbo) [reply src/editor.rs]: Neat.

**Comment by malbarbo** [comment]
I've update the format_size code.

**Comment by malbarbo** [comment]
Now travis is green!

**Pull request merged**